src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -527,6 +536,34 @@ String getLinker() {
         return "gcc";
     }
 
+    String getNativeImageCommand() {
+        return "native-image";
+    }
+
+    String getObjectFileExtension() {
+        return "o";
+    }
+
+    void specifyLinkProcessOutput(ProcessBuilder linkBuilder, String appName) {
+        linkBuilder.command().add("-o");
+        linkBuilder.command().add(getAppPath(appName));
+    }
+
+    void specifyLinkProcessLibraries(ProcessBuilder linkBuilder) {
+        linkBuilder.command().add("-ljava");
```

Comment:
Why not `addAll("", "", ...)`?
Also, I wouldn't expose the `ProcessBuilder`, but rather just return a `List<String>`, in line with other methods like `getTargetSpecificLinkFlags`